leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/CaliforniumEndpointsManager.java

```diff
@@ -198,8 +202,8 @@ public synchronized ServerIdentity createEndpoint(ServerInfo serverInfo) {
 
             // For bootstrap no need to have DTLS role exchange
             // and so we can set DTLS Connection as client only by default.
-            if (serverInfo.bootstrap && incompleteConfig.isClientOnly() == null) {
-                newBuilder.setClientOnly();
+            if (serverInfo.bootstrap && incompleteConfig.getDtlsRole() == null) {
+                newBuilder.set(DtlsConfig.DTLS_ROLE, DtlsRole.CLIENT_ONLY);
```

Comment:
I checked this specific setting. 
With the new Configuration, that's not longer possible to do that here depending on the user has overwritten a default or not. With all the other discussion, and the time already spend, and the time required to spend, lets ask:

Revert to the old Configuration?

Reply:
>  lets ask:  Revert to the old Configuration?

This is maybe too radical ?

The new Configuration/Attribute brings some real improvement. (typing, better dependency design, documentation)
But it also comes with behavior/paradigm changes and design changes.  

I feel this is not too surprising that when we try to integrate it we find some problematic points or design disagreement. 

My feeling is : 
- detecting changes
- discussing about  it
- searching solution
- getting agreement
- changing code if needed

All of this could be time consuming and that sounds "normal" to me.  (especially when we have some other tasks to handle at same time)

Maybe Leshan code need to be adapt to this new paradigm or californium must be enhanced to support Leshan needs or both. (Leshan or any other californium user by the way)

So I would keep the new design, but I'm not able to say it's ready now. (maybe after thinking more I would discover that just leshan adaptation is needed)
I guess I'm the only one who give you feedback about this. (I look at this deeply, thx to your PR)
I also guess others users could also raise some pertinent points. (like maybe @rogierc)
This could lead to more changes.

My concern here is more that you sounds to be : 
- hurry to release a 3.0.0 version. 
- don't really want to spend more time with this feature. 
- if I contribute code, you required there is no issue or any consequence for you.

which sounds not so so compliant with the situation. 

Personally, I need to review/integrate/test/polish lot of PRs about `CompositeObserve` first.
Then maybe I could find time to think/work more about this. 
I'm also absent from 24th August to the end of the month. (back the 1st)

Currently my concern: 
1. The static design (https://github.com/eclipse/leshan/pull/1073#discussion_r687709259  https://github.com/eclipse/leshan/pull/1073#discussion_r687860880 )
2. Unable to know if a value changed by user. (https://github.com/eclipse/leshan/pull/1073#discussion_r688277389)
3. `DtlsConfig` need `Configuration` (https://github.com/eclipse/leshan/pull/1073#discussion_r691051106)
4. Synchronizing 2 value from the config, maybe can be done with 2.  (https://github.com/eclipse/leshan/pull/1073#discussion_r687711109)


I need to think more about for all of this. (and maybe some are not real issue like 4.)

Reply:
> hurry to release a 3.0.0 version.

To be frank, for the old Configuration was not too bad. At least for 3.0 it is an alternative. I worked on the redesign, because my feeling was, that this changes the API a lot, and that's what happened. But if we fell uncomfortable with the changes it forces right now, postpone that to 4.0 will also work.

1. FMPOV, the static design helps a lot to have the definitions in place. A more dynamic one must care about that and that may result in other odd things.
2. If the user changes a value or not seems for me not that important, especially, if it's not known, if a value is kept by intention.
3. I'm not sure, to get this. In my intention, the Configuration is a "black-board", with all information in. The DtlsConfiguration takes that and adds some Callbacks, which are obviously not in the Configuration. As far as it works for me, in the very most cases there is one Configuration and that is used for all Endpoints and Connectors. 
4. That must be done by the user,  if it's really intended.

Reply:
Let me add:
What is a user intended to do in order to change a value?
Add/change it in Californium3.properties?

Reply:
May be, it should be clearer, what a user "means".
if it's a developer, I consider much knowledge, and complying to the pattern

```
  // Create CoAP Config
        Configuration coapConfig;
        File configFile = new File(Configuration.DEFAULT_FILE_NAME);
        if (configFile.isFile()) {
            coapConfig = new Configuration();
            coapConfig.load(configFile);
        } else {
            coapConfig = LeshanClientBuilder.createDefaultNetworkConfig();
            // here comes the user setup
            // coapConfig.set(...)
            coapConfig.store(configFile);
        }
```

does it. In the branch of the new Configuration, no other user's changes are available (at least, if don't assume queue of users.) so you don't need to check for them.

None developer users will in my opinion mainly edit the properties or provide parameters to the cli.